docs/quickstart/go.md

```diff
@@ -4,4 +4,184 @@ headline: Go Quick Start
+
+Now let's look at how to update the application with an extra method on the
+server for the client to call. Our gRPC service is defined using protocol
+buffers; you can find out lots more about how to define a service in a `.proto`
+file in [What is gRPC?](http://www.grpc.io/docs/#what-is-grpc) and [gRPC Basics:
+Go][]. For now all you need to know is that both the server and the client
+"stub" have a `SayHello` RPC method that takes a `HelloRequest` parameter from
+the client and returns a `HelloResponse` from the server, and that this method
+is defined like this:
+
+
+```protobuf
+// The greeting service definition.
+service Greeter {
+  // Sends a greeting
+  rpc SayHello (HelloRequest) returns (HelloReply) {}
+}
+
+// The request message containing the user's name.
+message HelloRequest {
+  string name = 1;
+}
+
+// The response message containing the greetings
+message HelloReply {
+  string message = 1;
+}
+```
+
+Let's update this so that the `Greeter` service has two methods. Make sure you are in the same examples dir as above (``` $GOPATH/src/google.golang.org/grpc/examples/helloworld```) 
+
+Edit `helloworld/helloworld.proto` and update it with a new `SayHelloAgain` method, with the same request and response
+types:
+
+```protobuf
+// The greeting service definition.
+service Greeter {
+  // Sends a greeting
+  rpc SayHello (HelloRequest) returns (HelloReply) {}
+  // Sends another greeting
+  rpc SayHelloAgain (HelloRequest) returns (HelloReply) {}
+}
+
+// The request message containing the user's name.
+message HelloRequest {
+  string name = 1;
+}
+
+// The response message containing the greetings
+message HelloReply {
+  string message = 1;
+}
+```
+
+## Generate gRPC code
+
+Next we need to update the gRPC code used by our application to use the new
+service definition. From the same examples dir as above (``` $GOPATH/src/google.golang.org/grpc/examples/helloworld```)
+
+```sh
+$ protoc -I helloworld/ helloworld/helloworld.proto --go_out=plugins=grpc:helloworld
+```
+This regenerates the helloworld.pb.go with our new changes.
+
+## Update and run the application
+
+We now have new generated server and client code, but we still need to implement
+and call the new method in the human-written parts of our example application.
+
+### Update the server
+
+Edit `greeter_server/main.go` and add the foloowing function to it:
+
+```
+func (s *server) SayHelloAgain(ctx context.Context, in *pb.HelloRequest) (*pb.HelloReply, error) {
+        return &pb.HelloReply{Message: "Hello again " + in.Name}, nil
+}
+```
+
+### Update the client
+
+Edit `greeter_client/main.go` to add the following code to the main function.
+
+```
+r, err = c.SayHelloAgain(context.Background(), &pb.HelloRequest{Name: name})
+if err != nil {
+        log.Fatalf("could not greet: %v", err)
+}
+log.Printf("Greeting: %s", r.Message)
+```
+
+### Run!
+
+1. Run the server 
+   ``` $ go run greeter_server/main.go
+
+2. On a different terminal, run the Client 
```

Comment:
In the rendered view, the next line is inlined. Probably need a blank line here...